deps/v8/src/arm/assembler-arm.cc

```diff
@@ -749,7 +778,7 @@ static bool fits_shifter(uint32_t imm32,
                          Instr* instr) {
   // imm32 must be unsigned.
   for (int rot = 0; rot < 16; rot++) {
-    uint32_t imm8 = (imm32 << 2*rot) | (imm32 >> (32 - 2*rot));
+    uint32_t imm8 = rot == 0 ? imm32 : ((imm32 << 2*rot) | (imm32 >> (32 - 2*rot)));
```

Comment:
Is this change really needed? It seems that both shift operations would zero-fill when applied to a `uint32_t`, and thus when `rot == 0`, this operation would be equivalent to `imm32 | 0`, or `imm32`, but I may be missing something.